### PR TITLE
fix: handle empty response in Bedrock AnthropicProvider

### PR DESCRIPTION
**Pull request by sansmoraxz**
# Description

Add bounds check as response may be empty for bedrock anthropic api

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py [hunk 1]
```diff
@@ -147,7 +147,9 @@ def get_text_from_stream_response(self, response: dict) -> str:
             return ""
 
     def get_text_from_response(self, response: dict) -> str:
-        return response["content"][0]["text"]
+        if response["content"]:
```
Comment (nerdai) [inline llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py:150]: I guess it's an Optional type?

nit (if this also works): `return response["content"][0]["text"] or ""`

**Review by nerdai — COMMENTED** [review]
Thanks @sansmoraxz -- left a minor comment in my review.

File: llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py (hunk 1 above)
Comment (sansmoraxz) [reply llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py:150]: No it does not. It was my earlier implementation, when testing privately.

the `content` array may be empty, and index of list 0 will give  'list index out of range'.

Comment (nerdai) [reply llama-index-integrations/llms/llama-index-llms-bedrock/llama_index/llms/bedrock/utils.py:150]: Got it. Thanks for the reply.

**Review by nerdai — APPROVED** [review]

**Comment by logan-markewich** [comment]
fyi @sansmoraxz have you tried the bedrock converse api yet? The implementation is much more simple, supports native function calling, and is likely less error prone

**Comment by sansmoraxz** [comment]
Sure. Will give it a try.

**Pull request merged**